url8556430df5_saulguillermo_agapi_tgz-pJ8-agapi_impl_ejemplos_IndividuoCombinatorioEjemploJ8/src/agapi/Configuracion.java

```diff
@@ -0,0 +1,452 @@
+ */
+public class Configuracion {
+
+	/**
+	 * Especifica al método {@code aTexto} devolver datos básicos de la última
+	 * generación de la última ejecución.
+	 */
+	public static final int ULTIMA_GENERACION = 0;
+
+	/**
+	 * Especifica al método {@code aTexto} devolver los datos básicos de todas
+	 * las generaciones de todas las ejecuciones.
+	 */
+	public static final int GENERACIONES_SIN_POBLACIONES = 1;
+
+	/**
+	 * Especifica al método {@code aTexto} devolver los datos de todas las
+	 * generaciones incluyendo sus respectivas poblaciones.
+	 */
+	public static final int GENERACIONES_CON_POBLACIONES = 2;
+
+	private Proceso proceso;
+
+	private boolean banderaIniciarProceso;
+	private boolean banderaNumeroEjecuciones;
+	private boolean banderaNumeroGeneraciones;
+	private boolean banderaSelector;
+	private boolean banderaSelectorPostCruce;
+	private boolean banderaProbabilidadCruce;
+	private boolean banderaProbabilidadMutacion;
+	private boolean banderaElitismo;
+	private boolean banderaTamanoPoblacion;
+	private boolean banderaTipoIndividuo;
+	private boolean banderaTamanoCromosoma;
+
+	/**
+	 * Devuelve un objeto de tipo {@link agapi.Proceso Proceso}. Este objeto se
+	 * crea el ejecutar el método {@code iniciarProceso}, por lo que devuelve el
+	 * {@linkplain agapi.Proceso proceso} asociado a esta configuración. A
+	 * partir de aquí se pueden obtener los demás datos de interés como
+	 * ejecuciones, generaciones, poblaciones, individuos...
+	 * 
+	 * @return el proceso asociado a esta configuración.
+	 */
+	public Proceso getProceso() {
+		if (banderaIniciarProceso) {
+			return proceso;
+		} else {
+			throw new IllegalStateException(
+					"El método iniciarProceso debe llamarse antes de getProceso");
+		}
+	}
+
+	/**
+	 * Inicia un proceso. Es decir, inicia una secuencia de
+	 * {@linkplain agapi.Ejecucion ejecuciones}. Este método tiene como
+	 * requisito, el previo llamado a los 10 <em>setters</em> que establecen los
+	 * parámetros de un algoritmo genético, de lo contrario dispara un excepción
+	 * {@code IllegalStateException}.
+	 * <p>
+	 * Se recomienda el llamado a los <em>setters</em> en el siguiente orden:
+	 * 
+	 * <blockquote>1. {@code setNumeroEjecuciones}</blockquote>
+	 * 
+	 * <blockquote>2. {@code setNumeroGeneraciones}</blockquote>
+	 * 
+	 * <blockquote>3. {@code setSelector}</blockquote>
+	 * 
+	 * <blockquote>4. {@code setSelectorPostCruce}</blockquote>
+	 * 
+	 * <blockquote>5. {@code setProbabilidadCruce}</blockquote>
+	 * 
+	 * <blockquote>6. {@code setProbabilidadMutacion}</blockquote>
+	 * 
+	 * <blockquote>7. {@code setElitismo}</blockquote>
+	 * 
+	 * <blockquote>8. {@code setTamanoPoblacion}</blockquote>
+	 * 
+	 * <blockquote>9. {@code setTipoIndividuo}</blockquote>
+	 * 
+	 * <blockquote>10. {@code setTamanoCromosoma}</blockquote>
+	 * 
+	 * @throws IllegalStateException
+	 *             si alguno de los 10 métodos obligatorios no ha sido llamado.
+	 */
+	public void iniciarProceso() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.